### PR TITLE
started to add tests for put/get

### PR DESCRIPTION
**Pull request by edhartnett**
Added a section to do a put and then a get.

But the get does not work. Not yet clear why so it is commented out while I work on it. But I want to merge at least the one put test I've added, to get the ball rolling.

This PR contains test code changes only. No library code is changed.

This is part of #379. Also part of #205.

I will merge to develop for testing.

### CODE REVIEW

File: tests/cunit/test_pioc.c [hunk 1]
```diff
@@ -446,6 +446,82 @@ int test_names(int iosysid, int num_flavors, int *flavor, int my_rank,
...
+        if ((ret = PIOc_enddef(ncid)))
+            ERR(ret);
+
+        /* Write some data. */
+        PIO_Offset start[NDIM] = {0, 0, 0};
+        PIO_Offset count[NDIM] = {1, 1, 1};
+        float data = 42.42;
+        if ((ret = PIOc_put_vara_float(ncid, varid, start, count, &data)))
+            ERR(ret);
+
```
Comment (jedwards4b) [inline tests/cunit/test_pioc.c]: Closing the file here and reopening it to read will solve the problem.

Comment (edhartnett) [inline tests/cunit/test_pioc.c]: I just discovered that. Why? Seems like a bug in pnetcdf. The other netcdf variants handle it fine, only pnetcdf fails.

Comment (jedwards4b) [inline tests/cunit/test_pioc.c]: Actually calling PIOc_sync(ncid) will solve the problem too.

Comment (jedwards4b) [inline tests/cunit/test_pioc.c]: In pnetcdf small writes are stored in memory until a sync or close file - or the memory buffer is exhausted - this is a performance feature.

**Pull request merged**